src/app/query/common/alex-sdk/alex-sdk.hooks.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useState } from 'react';
+
+import { AlexSDK, type Currency } from 'alex-sdk';
+import BigNumber from 'bignumber.js';
+
+import { logger } from '@shared/logger';
+import { createMoney } from '@shared/models/money.model';
+import { isDefined } from '@shared/utils';
+
+import { convertAmountToFractionalUnit } from '@app/common/money/calculate-money';
+import { pullContractIdFromIdentity } from '@app/common/utils';
+
+import { useAlexSdkLatestPricesQuery } from './latest-prices.query';
+import { useAlexSdkSwappableCurrencyQuery } from './swappable-currency.query';
+
+export function useAlexSdKCurrencyPriceAsMoney() {
+  const alexSDK = useState(() => new AlexSDK())[0];
+  const { data: supportedCurrencies = [] } = useAlexSdkSwappableCurrencyQuery(alexSDK);
+  const { data: prices } = useAlexSdkLatestPricesQuery(alexSDK);
+
+  return useCallback(
+    (principal: string) => {
+      if (!prices) {
+        logger.error('Latest prices could not be found');
+        return createMoney(0, 'USD');
```

Comment:
Why not return `null` here instead and have the return type `null | Money`? 

How can the UI differentiate between an actually worthless and failed to load state?

Reply:
This is bc the price is used later to determine the fiat value, I believe it would then need to accept a conditional prop but will check.